Provide monitoring/network stack solution for pocket-blockchain and pocket-validators

aqt01 opened the pull request:
Adds docker-compose for pocket-blockchain and pocket-validator with:

- Single pocket node validator
- Grafana/prometheus/loki/cadvisor/alertmanager for monitoring/metrics
- Reverse proxy, basic WAF and certbot auto certificate
- install instructions

Commit:
Adds pocket stack with pocket-blockchain

luyzdeleon commented on docker-compose/stacks/pocket-blockchain/docker-compose.yaml:
```diff
@@ -0,0 +1,143 @@
+      - POCKET_CORE_KEY=''
+    logging:
+      driver: loki
+        options:
```
Error in the indentation, should be at the same indentation as the `driver`.

aqt01 replied:
Thanks! @luyzdeleon fixed in https://github.com/pokt-network/pocket-core-deployments/pull/170/commits/aa859c26bc851e232ce8085770de044b120d3ac8

Commit:
Fix ingest of tendermint/pocket for prometheus, adds latest beta version for pocket and clarification on docs

aqt01 committed:
Fix indent typo on driver logging

aqt01 committed:
Add correct proxy_pass to pocket node

aqt01 committed:
Fix nginx proxy routing and removed genesis mapping because pocket already downloads it

aqt01 committed:
Add more doc details on README.md. Deactivated :info logs

aqt01 committed:
Adds pocket-validator stack and updates README instructions

aqt01 committed:
Add hardware requirements

julio-jimenez commented on docker-compose/stacks/pocket-blockchain/README.md:
```diff
@@ -0,0 +1,86 @@
+Disk: 100GB is the minimum HDD requirement
+
+For more information, see: 
+https://docs.pokt.network/docs/before-you-dive-in#hardware-requirements
```
We should encourage people to use SSD instead of HDD

aqt01 replied:
Totally, i will edit this and also request to edit those in the docs!  Thanks! @julio-jimenez

julio-jimenez commented on docker-compose/stacks/pocket-blockchain/README.md:
```diff
@@ -0,0 +1,86 @@
+>  sudo bash install.sh
+
+
+2. Change the following settings according your needs:
```
I think we can skip this part since this should only be done for validators. If this is done, then the user will provide free infrastructure to the public.

aqt01 replied:
Totally, this was just an example for setting up basic pocket-core-blockchain but yes, i'm exposing this. I will add the ethereum stack and the basic auth nginx config, so it can be a stack for current supported blockchains. Thanks for the review @julio-jimenez

julio-jimenez commented on docker-compose/stacks/pocket-blockchain/README.md:
```diff
@@ -0,0 +1,86 @@
+    - The env variable `GF_SECURITY_ADMIN_PASSWORD` in docker-compose grafana service. 
+
+
+3. Configure your pocket node by editing on `node1/config.json` the following variables:
```
We could use the default config created when pocket starts instead of having to create/edit a new one.

julio-jimenez commented on docker-compose/stacks/pocket-blockchain/README.md:
```diff
@@ -0,0 +1,86 @@
+> nslookup monitoring.${DOMAIN}
+
+
+6. Proceed to validate the ssl certificate by doing 
```
No need if not exposing this node to the public

julio-jimenez commented on docker-compose/stacks/pocket-blockchain/docker-compose.yaml:
```diff
@@ -0,0 +1,139 @@
+      - web
+
+  node1:
+    image: poktnetwork/pocket-core:Beta-0.5.2.3
```
Use RC- instead of Beta- image?

julio-jimenez commented on docker-compose/stacks/pocket-blockchain/docker-compose.yaml:
```diff
@@ -0,0 +1,139 @@
+
+  node1:
+    image: poktnetwork/pocket-core:Beta-0.5.2.3
+    container_name: node1 
```
While they are in two different compose files, we should specify that this is a not a validator. Add something like `pocket-blockchain-1`

julio-jimenez commented on docker-compose/stacks/pocket-validator/README.md:
```diff
@@ -0,0 +1,99 @@
+> cp ./node_data/node_key.json node1/ 
+
+
+9. Just do `docker-compose down && docker-compose up -d ` and you will be up & running! 
```
Why `docker-compose down` before `docker-compose up -d`?

julio-jimenez commented on docker-compose/stacks/pocket-validator/README.md:
```diff
@@ -0,0 +1,99 @@
+
+CPU: 4 CPU’s (or vCPU’s)
+Memory: 8 GB RAM
+Disk: 100GB is the minimum HDD requirement
```
Let's encourage SSD instead of HDD

julio-jimenez reviewed (commented):
Looks great! Just a few minor details

nelcolon committed:
Adding naxsi rules for WAF solution

Added Naxsi core set rules for WAF into Nginx.

aqt01 committed:
Improve readme.md. Replaced /root/pocket-core-deployments/docker-compose/stacks by ./

aqt01 committed:
Add proxy timeout to 260 s to support long geth queries

aqt01 committed:
Add vim files for .gitignore

aqt01 committed:
Add requirement of docker and docker-compose version

aqt01 committed:
remove copy of config files to nginx

aqt01 committed:
Fix some docs issues. Also add .env for example purposes

aqt01 committed:
Fix typo for docker-compose

aqt01 committed:
Updated nginx similar to docker alpine

aqt01 committed:
Fix typo for docker-entrypoint.sh

aqt01 committed:
Add permission for running .sh

aqt01 committed:
removing naxsi rule from certbot validation

aqt01 committed:
Add section for adding passphrase and add default username for login on grafana

aqt01 committed:
Fix path on pocket-validator and add missing empty env variables

nelcolon committed:
Updating the README.md

Implemented better markdown and did some improvements in headings and code blocks.

nelcolon committed:
Delete naxsi core rules

aqt01 committed:
Removed naxsi rules

aqt01 commented:
Closes 
https://github.com/pokt-network/pocket-core-deployments/issues/177 
https://github.com/pokt-network/pocket-core-deployments/issues/172
https://github.com/pokt-network/pocket-core-deployments/issues/171
https://github.com/pokt-network/pocket-core-deployments/issues/175
